### PR TITLE
Check the latest version of @eclipse-che/api

### PR DESCRIPTION
**Pull request by evidolob**
And publish only when it's don't match

### What does this PR do?
Fix publishing npm package with che dto d.ts file, during che release

### What issues does this PR fix or reference?
#12454

### CODE REVIEW

File: typescript-dto/publish.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env sh
```
Comment (benoitf) [inline typescript-dto/publish.sh:1]: missing copyright headers

File: typescript-dto/publish.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+#get the latest version of @eclipse-che/api
+api_version=$(yarn -s info @eclipse-che/api version)
+
+#publish only if latest version doesn't match current version
+if [ $api_version != $1 ];
```
Comment (benoitf) [inline typescript-dto/publish.sh]: use of double quotes (https://www.shellcheck.net/)

File: typescript-dto/publish.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env sh
+
+#get the latest version of @eclipse-che/api
+api_version=$(yarn -s info @eclipse-che/api version)
```
Comment (benoitf) [inline typescript-dto/publish.sh:16]: we should have

```
set -e
set -u
```

 as well

should we first check that yarn tool is installed as well ?

**Review by riuvshin — APPROVED** [review]

**Review by benoitf — APPROVED** [review]

File: typescript-dto/publish.sh (hunk 3 above)
Comment (evidolob) [reply typescript-dto/publish.sh:16]: I added 
```
set -e
set -u
```
No, as that script called from docker which already have `yarn` https://github.com/eclipse/che/blob/b818fccea3496f85579ad1affaf880674bb718f1/typescript-dto/Dockerfile#L25

File: typescript-dto/publish.sh (hunk 1 above)
Comment (evidolob) [reply typescript-dto/publish.sh:1]: Done

**Pull request merged**